DSharpPlus.Interactivity/InteractivityExtension.cs

```diff
@@ -753,13 +754,25 @@ public async Task SendPaginatedResponseAsync(DiscordInteraction interaction, boo
                     bts.SkipRight.Disable();
             }
 
-            var builder = new DiscordInteractionResponseBuilder()
-                .WithContent(pages.First().Content)
-                .AddEmbed(pages.First().Embed)
-                .AsEphemeral(ephemeral)
-                .AddComponents(bts.ButtonArray);
+            if (asEditResponse)
+            {
+                var builder = new DiscordWebhookBuilder()
+                    .WithContent(pages.First().Content)
+                    .AddEmbed(pages.First().Embed)
+                    .AddComponents(bts.ButtonArray);
 
-            await interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);
+                await interaction.EditOriginalResponseAsync(builder);
```

Comment:
If I'm not mistaken, this returns `Task<DiscordMessage>` so what you can do is actually something along the lines of

```cs
DiscordMessage message = null;

if (asEditResponse)
    message = await interaction.EditOriginalResponseAsync(builder);
else 
    await interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);

message ??= await interaction.GetOriginalResponseAsync();
```

Reply:
`EditOriginalResponseAsync` and `CreateResponseAsync` uses different builder and `DiscordWebhookBuilder` can't be casted to `DiscordInteractionResponseBuilder` and vice versa so merging both builder seems not possible.